server/backends/blobstore/util/util.go

```diff
@@ -121,3 +127,82 @@ func RecordDeleteMetrics(typeLabel string, startTime time.Time, err error) {
 		metrics.BlobstoreTypeLabel: typeLabel,
 	}).Observe(float64(duration.Microseconds()))
 }
+
+type compressedWriter struct {
+	ctx       context.Context
+	committer interfaces.Committer
+	zw        io.WriteCloser
+	closer    io.Closer
+	label     string
+	n         int
+}
+
+func NewCompressedBlobStoreWriter(
+	ctx context.Context,
+	committer interfaces.Committer,
+	writer io.Writer,
+	closer io.Closer,
+	label string,
+) interfaces.CommittedWriteCloser {
+	return &compressedWriter{
+		ctx: ctx, committer: committer,
+		zw:     newCompressWriter(writer),
+		closer: closer,
+		label:  label,
+		n:      0,
+	}
+}
+
+func (c *compressedWriter) Commit() error {
+	c.zw.Close()
+	return c.committer.Commit()
+}
+
+func (c *compressedWriter) Write(p []byte) (int, error) {
+	//Unnecessary without the RecordWriteMetrics call
+	// start := time.Now()
+	_, spn := tracing.StartSpan(c.ctx)
```

Comment:
This is too fine grained for a span. Normally a span represents a slightly higher level operation (i.e. "write blob") and individual writes might be represented as span events.

Reply:
If you do want to bring it back in the future, the way to do it is create a new span in `Writer` and then call `span.AddEvent` for each individual write. This allows you to add fine grained timing information w/o creating new spans.